### ISSUE TITLE
Replace Form with FormBuilder

### ISSUE THREAD

**Issue by nicolayr** [dhis2/settings-app#47]
The `DataApprovalLevels` and `Oauth2Clients` components are still using the `Form` component. This should be replaced with the `FormBuilder` component.

**Comment by adhbh** [comment]
@nicolayr  @Markionium  
I have implemented this for Oauth here: https://github.com/adhbh/settings-app/commit/e8475f4575238583c1a4e808e292f7100099f7d6

However before opening PR, i have a few doubts -
1.  I have used FormBuilder to create Save, Cancel and Delete buttons as well. Is form builder supposed to build "RaisedButton" and "FlatButton" from material-ui as well?
2. If yes, then in FormBuilder,  shouldn't there be a prop to modify inline style of the container div? 

Since the container div has predefined style(ie. position: relative), the buttons are not properly alligned as per the requirements (see screenshot)

If Save, Cancel buttons are not supposed to be build by FormBuilder, and can be added separately, then there is no issue. However I don't think that is a good practice. Hence there should be a prop for FormBuilder to modify style of container div. (it should add display: 'inline-block' in this case for proper alignment)

Please let me know what you think.
I am also separating workflow level and approval level and changing the FormBuilder for them as well. A similar kind of problem will arise there as well.

<img width="1156" alt="screen shot 2016-03-25 at 1 40 24 am" src="https://cloud.githubusercontent.com/assets/6566723/14030208/7ef216a4-f22c-11e5-99d9-73a3b1bdd929.png">

**Comment by nicolayr** [comment]
Nice work @adhbh !

There are two primary reasons to use `FormBuilder` at the moment: To dynamically generate forms (as in the settings app) and to facilitate validation of the form fields. Since `FormBuilder` never actually renders a HTML `<form>` tag, it's not really a form in the traditional sense. If it was a traditional form I would agree that rendering the buttons outside the form would be bad practice. But unless I'm missing something that doesn't really apply here?

In short, since they aren't really form fields, I would move the buttons outside the `FormBuilder` component.

**Comment by adhbh** [comment]
@nicolayr 
Yes, I agree with you and for now I will move buttons outside FormBuilder. But I got this doubt when I saw the postbutton in Email setting category is created using FormBuilder. Check https://github.com/dhis2/settings-app/blob/master/src/app.component.js#L192
